server/room.ts

```diff
@@ -266,45 +186,51 @@ export class Room {
   getHostState() {
     return {
       video: this.video,
+      videoDuration: this.videoDuration,
       videoTS: this.videoTS,
       paused: this.paused,
     };
   }
 
-  async resetRoomVM() {
-    const id = this.vBrowser && this.vBrowser.id;
-    this.vBrowser = undefined;
-    this.roster.forEach((user, i) => {
-      this.roster[i].isController = false;
-    });
-    this.cmdHost(undefined, '');
-    if (id) {
-      try {
-        await resetVM(id);
-      } catch (e) {
-        console.error(e);
-      }
+  cmdHost(
+    socket: Socket,
+    data?: PlaylistVideo,
+    options = { skipMessage: false }
+  ) {
+    if (!socket) {
+      return;
+    }
+
+    if (data) {
```

Comment:
I assume this is to handle the case where someone sets the current video to something that's already somewhere in the queue. Not sure if we have to necessarily remove it if that happens.